HPCC-13021 Fix CMake 3.x warnings

xwang2713 opened the pull request:
Replace GET_TARGET_PROPERTY(..., LOCATION) with $&lt;TARGET_FILE:variable_name&gt; (CMake CMP0026)

Remove Clienttools dependencies on targets with prefix "ws_".

@Michael-Gardner please review

Signed-off-by: Xiaoming wang xiaoming.wang@lexisnexis.com

Commit:
HPCC-13021 Fix CMake 3.x warnings

CMP0026 Replace GET_TARGET_PROPERTY(..., LOCATION) with $<TARGET_FILE:variable_name>
CMP0046 Remove non-existent target by moving  hidl and esdlcmd-xml above esdlcmd in CMakeLists.txt
Change CMake miniumu version from 2.8.3 to 2.8.11

AttilaVamos commented:
Automated Smoketest Build: success
HPCC Start: OK
HPCC Stop: OK

Michael-Gardner commented:
@xwang2713 I'm getting the @PROTECTED_AT@ issue again, using 3.2.2.  I could have sworn they reverted that patch in 3.2.0 so I don't know if this is a regression, or if it didn't make it into the final release.  I'll check tomorrow in the morning.

xwang2713 commented:
@PROTECTED_AT@ will be fixed on CMake 3.3.
http://www.cmake.org/Bug/view.php?id=14782#c35227

richardkchapman commented on ecl/ecl-package/CMakeLists.txt:
```diff
@@ -56,7 +56,12 @@ include_directories (
-add_dependencies ( ecl-packagemap espscm ws_packageprocess )
+
+set( WS_PACKAGEPROCESS_DEP )
+if( NOT CLIENTTOOLS_ONLY )
```
This makes no sense. I can't be dependent on something only if I am doing a full build but not if I am only building the client tools.

I suspect what is really happening is that the client tool is dependent on the scm file of the service that it calls (which is fair enough).

xwang2713 replied:
Builds are OK with old cmake 2.8.11+.

Currently ws_packageprocess and ws_workunits are only build with "Platform" which is why there are not available in "Clienttools-Only" build.

Guess I should make ws_packageprocess and ws_workunits build on "Clienttools" which is enabled with "Platform" build.

richardkchapman replied:
That would be a workaround but the correct fix is to state the dependency that actually exists - i.e. a dependency on the scm file

Is there a reason the relevant scm files are not included in the espscm target?

richardkchapman replied:
Our cmake file states

 cmake_minimum_required (VERSION 2.8.3)

If we are now requiring 2.8.11, we need to change that. We may also need to update some instructions on wiki pages regarding when/whether the default cmake for a distro is modern enough.

xwang2713 replied:
I will install CMake 2.8.3 and try it. All our build systems have CMake 2.8.11+.

For dependencie there are the ones from scm, for example ws_workunits.ecm, ws_packageprocess.ecm. These are fine since scm built with Clienttools and espscm is added to the targets' dependencies. The problem comes from esp/service/<ws_workunits|ws_packageprocess> which are only built with "Platform" currently. Clienttools eclcmd, etc depend on esp/service/<ws_workunits|ws_packageprocess>.

richardkchapman replied:
I don't know what esp/service is

richardkchapman replied:
I think the dependency here is just wrong.

@afishbeck Do you know anything about this?

I not also that there are similar issues in 

./ecl/eclcmd/queries/CMakeLists.txt:add_dependencies ( ecl-queries espscm ws_workunits )
./ecl/eclcmd/roxie/CMakeLists.txt:add_dependencies ( ecl-roxie espscm ws_workunits )

afishbeck replied:
The client tools should just depend on the client code from the ecm files they use being built.  I'd have thought it was covered by "espscm" which should build all of the ecm files?

richardkchapman replied:
That's what I thought. These CMake files seem to have had this extra (unwanted) dependency in from when they were created - I think the correct fix is to remove them.

xwang2713 replied:
Remove ws_xxx dependencies from clienttools components.
Tried CMake 2.8.3 with the fix CMP0026 but it doesn't work. 
 I set HPCC Platform and LN minimum CMake version to 2.8.11 in the top level CMakeLists.txt files. I didn't try CMake version between 2.8.3 and 2.8.11.

Will update HPCC github CMake Version if the old version is set.

AttilaVamos commented:
Automated Smoketest Build: success
HPCC Start: OK
HPCC Stop: OK

AttilaVamos commented:
Automated Smoketest Build: success
HPCC Start: OK
HPCC Stop: OK

richardkchapman commented:
There's a typo in the commit message.

Are there any distros that we target that ship with a version of cmake between 2.8.3 and 2.8.10 in the default repository? If so we'll need to update our build machines (and instructions) for those distros to indicate that CMake needs to be installed manually rather than from the distro repository.

AttilaVamos commented:
Automated Smoketest Build: success
HPCC Start: OK
HPCC Stop: OK

xwang2713 commented:
Corrected and updated commit message.

If installed from distro repositoriy, CentOS 5,6,Ubuntu 10.04 (will drop support soon), 12.04 have older CMake version than 2.8.11. User need manually install the required CMake . I can work with Jim to update HPCC Github wiki "Build HPCC"/CMake section.

All our build systems have CMake 2.8.11 or later.

richardkchapman commented:
Merged manually (correcting typo in commit message) into 5.4.0
